### PR TITLE
Fix(Pegins): Return an error to invalidate pegin v1 with invalid ref block hash

### PR DESCRIPTION
**Pull request by SatwikPrabhu**
Fix for https://github.com/botanix-labs/botanix/issues/996

Also moved the test case for invalid ref block hash in `test_invalid_pegin` to the front of the invalid pegin list as it essentially replicates the edge case encountered in https://github.com/botanix-labs/Macbeth/pull/664#discussion_r2048381145

### CODE REVIEW

**Review by lamafab — DISMISSED** [review]
Interesting; what is the current behavior if the block hash is not found?

**Review by rwlockbg — DISMISSED** [review]

File: crates/ethereum/evm/src/execute.rs [hunk 1]
```diff
@@ -489,6 +489,13 @@ where
                                 })?;
                             bitcoin_checkpoint = package.bitcoin_checkpoint;
                         }
+                        else {
+                            return Err(MintContractError::InvalidPeginData {
+                                error: "No block found for reference block hash".to_string(),
```
Comment (arminsabouri) [inline crates/ethereum/evm/src/execute.rs]: Any reason this error message is different that the one above?
https://github.com/botanix-labs/Macbeth/pull/673/files#diff-7fbb0c1e65c92e8879aa308fee2538af673a2cd6d70f9b5b1103cf4d6554604dL473

They seem to indicate the same thing? I think the above should represent that an internal db error occured and this one is no block found. 

If they are truly the same case (no block found) then we should consolidate error messages.

**Review by arminsabouri — CHANGES_REQUESTED** [review]
One nitpick about error messaging

**Comment by SatwikPrabhu** [comment]
> Interesting; what is the current behavior if the block hash is not found?

The current behavior uses the bitcoin checkpoint that would have been used in a v0 pegin corresponding to the tip of the bitcoin blockchain and would have assumed that the block headers list would also have extended upto the tip.

File: crates/ethereum/evm/src/execute.rs (hunk 1 above)
Comment (SatwikPrabhu) [reply crates/ethereum/evm/src/execute.rs]: Done!

**Review by rwlockbg — DISMISSED** [review]

File: bin/test-suite/src/suite/consensus/invalid_transactions/test_invalid_pegin.rs [hunk 2]
```diff
@@ -38,6 +38,24 @@ async fn generate_invalid_pegin_metas(
...
+                gateway_address_response.aggregate_public_key.as_str(),
+            )
+            .expect("valid public key"),
+            tx: pegin_tx.clone(),
+            merkle_proof: pmt.clone(),
+            block_headers: headers.clone(),
+        },
+        ref_block_hash: B256::from_slice(&[0; 32]),
+    })];
+    invalid_pegin_meta_cases.push((invalid_ref_hash_meta, "Invalid reference hash for v1"));
```
Comment (arminsabouri) [inline bin/test-suite/src/suite/consensus/invalid_transactions/test_invalid_pegin.rs]: Nit: refrence hash implies v1. 

```suggestion
    invalid_pegin_meta_cases.push((invalid_ref_hash_meta, "Invalid reference hash"));
```

File: crates/ethereum/evm/src/execute.rs [hunk 3]
```diff
@@ -471,7 +471,7 @@ where
                         if let Some(block) = provider
                             .find_block_by_hash(hash, reth_provider::BlockSource::Any)
                             .map_err(|_| MintContractError::InvalidPeginData {
-                                error: "Reference block hash not found".to_string(),
+                                error: "Database error fetching reference block hash".to_string(),
```
Comment (arminsabouri) [inline crates/ethereum/evm/src/execute.rs]: Shouldnt this be a panic condition instead of an event that causes a revert? If the database is corrupt and we can't validate pegins then I would think we expect. @scottmillner can you check my reasoning on this?

Comment (SatwikPrabhu) [reply crates/ethereum/evm/src/execute.rs]: This would cause a revert with the logic introduced in https://github.com/botanix-labs/Macbeth/pull/664 right?

File: bin/test-suite/src/suite/consensus/invalid_transactions/test_invalid_pegin.rs (hunk 2 above)
Comment (SatwikPrabhu) [reply bin/test-suite/src/suite/consensus/invalid_transactions/test_invalid_pegin.rs]: Done!

File: crates/ethereum/evm/src/execute.rs (hunk 3 above)
Comment (arminsabouri) [reply crates/ethereum/evm/src/execute.rs]: The question is whether or not it should

Comment (scottmillner) [reply crates/ethereum/evm/src/execute.rs]: Thanks for pointing this out @0xBEEFCAF3 

I think the answer is this:
- We only want to revert when the sender submitted a tx that fails validation
- Not if we have an io error (db, call to btc-server, etc) bc that's something wrong on our end not with the tx

So, I agree we should panic if we have a database error

Comment (SatwikPrabhu) [reply crates/ethereum/evm/src/execute.rs]: Done!

**Review by scottmillner — APPROVED** [review]
